src/group_operations/multiplication_operation.jl

```diff
@@ -0,0 +1,295 @@
+function Base.:\(
+    e::Identity{<:AbstractMultiplicationGroupOperation},
+    ::Identity{<:AbstractMultiplicationGroupOperation},
+)
+    return e
+end
+
+_doc_compose_mult = """
+    compose(G::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, g, h)
+    compose!(G::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, k, g, h)
+
+Compute the group operation composition of `g` and `h` with respect to
+an [`AbstractMultiplicationGroupOperation`](@ref) on `G`, which falls back to calling
+`g*h`, where `*` is assumed to be overloaded accordingly.
+
+This can be computed in-place of `k`.
+"""
+
+@doc "$(_doc_compose_mult)"
+compose(::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, g, h) where {𝔽}
+
+@doc "$(_doc_compose_mult)"
+compose!(::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, k, g, h) where {𝔽}
+
+function _compose!(::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, k, g, h) where {𝔽}
+    # perform the multiplication “safe”, that is, even when providing
+    # one of the inputs `g,h`` and as output `k`
+    (k === g || k === h) ? copyto!(k, g * h) : mul!(k, g, h)
+    return k
+end
+
+Base.inv(e::Identity{<:AbstractMultiplicationGroupOperation}) = e
+
+LinearAlgebra.det(::Identity{<:AbstractMultiplicationGroupOperation}) = true
+
+_doc_diff_conjugate_add = """
+    diff_conjugate(G::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, g, h, X)
+    diff_conjugate!(G::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, Y, g, h, X)
+
+Compute the differential of the conjutage ``c_g(h) = g$(_math(:∘))h$(_math(:∘))g^{-1} = ghg^{-1}``,
+which simplifies for an [`AbstractMultiplicationGroupOperation`](@ref) to ``D(c_g(h))[X] = gXg^{-1}``.
+"""
+
+@doc "$(_doc_diff_conjugate_add)"
+diff_conjugate(G::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, g, h, X) where {𝔽}
+
+@doc "$(_doc_diff_conjugate_add)"
+function diff_conjugate!(
+    G::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation}, Y, g, h, X
+) where {𝔽}
+    inv_right_compose!(G, Y, X, g) # Y = Xg^{-1}
+    compose!(G, Y, g, Y) # Y = gY
+    return copyto!(LieAlgebra(G), Y, X)
+end
+
+_doc_diff_inv_mult = """
+    diff_inv(G::LieGroup{𝔽,<:AbstractMultiplicationGroupOperation},, p, X)
+
+Compute the value of differential ``Dι_{$(_math(:G))}(g)[X] of matrix inversion ``ι_{$(_math(:G))}(g) := g^{-1}`` at ``X ∈ 𝔤``
+in the [`LieAlgebra`](@ref) ``𝔤`` of the [`LieGroup`](@ref) `G`.
+
+The formula is given by
+
+```math
+Dι_{$(_math(:G))}(g)[X] = -g^{$(_tex(:transp))}Xp^{-1},
+```
+
+which stems from using the differential of the inverse from [Giles:2008](@cite) given by
+``D(g^-1)[X] = -g^{-1}Xg^{-1}`` composed with the push forward of the left composition
+``Dλ_$(_math(:e))(g)[X] = gX`` mapping from the Liea algebra into the tangent space at ``g``,
+and its adjoint ``D^*λ_$(_math(:e))(g)[X] = g^{$(_tex(:transp))}X``.
+Then we get ``g^{$(_tex(:transp))}(g^{-1}(gX)g^{-1})`` which simplifies to ``-g^{$(_tex(:transp))}Xp^{-1}`` from above.
+"""
```

Comment:
@mateuszbaran could you maybe verify / check this? I think this should be correct. I only mention the Lie algebra case, because for now we assume all Lie groups represent their tangents in the Lie algebra (we can handle others when they appear). 
And I would even prefer having a reference for that derivation.

The one thing I do not like in this is, that the adjoint is only correct if the multiplicative Lie group we consider here has the Euclidean metric in the Lie algebra.

Reply:
Maybe a generic `adjoint(𝔤::LieAlgebra, X)` could resolve the metric/adjoint problem.

Reply:
I think I understood where its from, I just have to finde a good place to write this down.

If we have $Df(g) = L$ as a differential (linear map L) we have to change  _both_ “input” and “output” of that map, so if we write that out we can push forward any vector (from e to g before applying L) but we have to do the adjoint on the result (to pull back from g to e) afterwards.

We should try to write this down somewhere central maybe, or find a good reference.

Reply:
I remember looking for a good reference for this but I couldn't find it.

> The one thing I do not like in this is, that the adjoint is only correct if the multiplicative Lie group we consider here has the Euclidean metric in the Lie algebra.

I think we can always have a specialized method for those non-Euclidean-metric cases.

> If we have D f ( g ) = L as a differential (linear map L) we have to change _both_ “input” and “output” of that map, so if we write that out we can push forward any vector (from e to g before applying L) but we have to do the adjoint on the result (to pull back from g to e) afterwards.

IIRC that's how it goes, yes.

Reply:
Yes, we can surely leave this as is, then and specials / refine later.
One could do an `adjoint(...)` method for this case when refining this.

I also already documented that the tangent vectors for general linear are now in the Lie algebra and no linger in the embedding.